src/tasks/batch_integration/methods/mnn/config.vsh.yaml

```
@@ -0,0 +1,40 @@
+# use method api spec
+__merge__: ../../api/comp_method_feature.yaml
+functionality:
+  name: mnn
+  info:
+    label: MNN
+    summary: "Batch effect correction by matching mutual nearest neighbors, Python implementation."
+    description: |
+      "Large-scale single-cell RNA sequencing (scRNA-seq) data sets that are produced in different laboratories and at different times contain batch effects that may compromise the integration and interpretation of the data. Existing scRNA-seq analysis methods incorrectly assume that the composition of cell populations is either known or identical across batches. We present a strategy for batch correction based on the detection of mutual nearest neighbors (MNNs) in the high-dimensional expression space. Our approach does not rely on predefined or equal population compositions across batches; instead, it requires only that a subset of the population be shared between batches. We demonstrate the superiority of our approach compared with existing methods by using both simulated and real scRNA-seq data sets. Using multiple droplet-based scRNA-seq data sets, we demonstrate that our MNN batch-effect-correction method can be scaled to large numbers of cells."
+    reference: "hie2019efficient"
+    repository_url: "https://github.com/chriscainx/mnnpy"
+    documentation_url: "https://github.com/chriscainx/mnnpy#readme"
+    v1:
+      path: openproblems/tasks/_batch_integration/batch_integration_graph/methods/mnn.py
+      commit: 29803b95c88b4ec5921df2eec7111fd5d1a95daf
+    preferred_normalization: log_cpm
+    variants:
+      mnn_full_unscaled:
+      mnn_hvg_unscaled:
+        hvg: true
+      mnn_hvg_scaled:
+        hvg: true
+        preferred_normalization: log_cpm_scaled
+      mnn_full_scaled:
+        preferred_normalization: log_cpm_scaled
+  resources:
+    - type: python_script
+      path: script.py
+platforms:
+  - type: docker
+    image: ghcr.io/openproblems-bio/base_r:1.0.0
+    setup:
+      - type: apt
+        packages: [ gcc-8 ]
+      - type: python
+        pypi:
+          - mnnpy
+          - scanpy
+          - scib==1.1.3
```

i've been debugging and found that the following should work.

```suggestion
  - type: docker
    image: eddelbuettel/r2u:20.04
    setup:
      - type: apt
        packages:
          - procps
          - git
          - libhdf5-dev
          - libgeos-dev
          - python3.8
          - python3-pip
          - python3.8-dev
          - python-is-python3
      - type: python
        pypi:
          - rpy2
          - anndata~=0.8.0
          - scanpy
          - pyyaml
          - requests
          - jsonschema
          - scib==1.1.3
        github: chriscainx/mnnpy
      - type: r
        packages:
          - anndata
          - BiocManager
        github:
          - rstudio/reticulate